### DISCUSSION TITLE
Using org-ql-find to select the target for a link to insert

### DISCUSSION THREAD

**Discussion by CyberFox001** [alphapapa/org-ql#479] [Ideas]
Hello,

Sometimes, while I write text into an Org-mode buffer, I need to insert a link to another Org-mode entry.

For now, I:

- Use `org-ql-find` to find the target of my link and display it
- Store the location of my target with `org-store-link`
- Come back to my previous location
- Insert a link with `org-insert-link`

It would be nice if it can be done in less steps. 

For example, a function named `org-ql-insert-link` that let user select the link target with same UI as org-ql-find, but after the target selection the link is directly inserted at Emacs point.

What do you think ?

**Comment by alphapapa** [comment]
This is handled well with Embark.  You can `C-.` on an item in the completion list and choose to store a link to it.

**Comment by CyberFox001** [comment]
How, I didn't know Embark. I will take a look. Thank you.

But it store the link and user still have to insert it.

The idea is to avoid this step and insert the link directly.

**Reply by alphapapa** [reply]
Then you should probably write your own command that both stores and inserts the link, and use something like Embark to activate it.

**Reply by CyberFox001** [reply]
Is a org-ql internal function exist to let user select a target and then return the path to the selected target ?

**Reply by alphapapa** [reply]
> Is a org-ql internal function exist to let user select a target and then return the path to the selected target ?

I don't understand your question.  But all of the source code is there to be explored.  There aren't that many `defun`s in `org-ql`; `M-x apropos-function RET`.
